charts/kargo/templates/api/deployment.yaml

```diff
@@ -59,9 +59,19 @@ spec:
           image: {{ include "kargo.image" . }}
           imagePullPolicy: {{ .Values.image.pullPolicy }}
           command: ["/usr/local/bin/kargo", "api"]
-          {{- with (concat .Values.global.env .Values.api.env) }}
           env:
-            {{- toYaml . | nindent 10 }}
+          - name: GOMEMLIMIT
+            valueFrom:
+              resourceFieldRef:
+                containerName: api
+                resource: limits.memory
+          - name: GOMAXPROCS
+            valueFrom:
+              resourceFieldRef:
+                containerName: api
+                resource: limits.cpu
```

Comment:
Think this may need `if` wrapping to ensure `limits.memory` is set?

Reply:
I tried it. With no limits set, they come through as all CPU and memory on the node.